src/ir/stack-utils.h

```diff
@@ -75,6 +81,10 @@ namespace StackUtils {
 // Iterate through `block` and remove nops.
 void removeNops(Block* block);
 
+// Whether `expr` may be unreachable in Poppy IR. True for control flow
+// structures and polymorphic unreachable instructions.
+bool mayBeUnreachable(Expression* expr);
```

Comment:
I still think my comment about this is relevant: https://github.com/tlively/binaryen/pull/1#discussion_r483996005

The comment here doesn't clearly say which of the two cases it is IMO. If it is "not exited normally" or such, that might be good to clarify?

Reply:
I mean, is "unreachable" here referring to a type in the type system, or to a particular property, etc.